libsplinter/src/service/scabbard/state.rs

```diff
@@ -219,7 +225,7 @@ impl ScabbardState {
 
         // Get the results and shutdown the scheduler
         let batch_result = result_rx
-            .recv_timeout(std::time::Duration::from_secs(EXECUTION_TIMEOUT))
+            .recv_timeout(Duration::from_secs(EXECUTION_TIMEOUT))
```

Comment:
recv_timeout is panic-prone

Reply:
This isn't changed in this PR, just added the import for `std::time::Duration`

Reply:
That may be true, but you use it in your iterator later.

Reply:
Created a story to address this issue for splinter as a whole